### PR TITLE
Regular expression changes

### PR DESCRIPTION
**Pull request by peco2282**
## Summary
Simple modification of regular expressions.
change regular expression format.
like this...

`[0-9]` -> `\d`
`[A-Za-z0-9\_]` -> `\w`

[You can check it here.](https://regex101.com/r/KRV4tW/1)

(If you are doing this intentionally, you may close it.)

## Checklist

- [x] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...)

### CODE REVIEW

**Comment by Lulalaby** [comment]
@Pycord-Development/maintainers what you think

**Comment by peco2282** [comment]
I deleted the incorrectly entered characters when making the change.
Sorry.

File: discord/webhook/sync.py [hunk 1]
```diff
@@ -705,7 +705,7 @@ def from_url(cls, url: str, *, session: Session = MISSING, bot_token: Optional[s
             A partial webhook is just a webhook object with an ID and a token.
         """
         m = re.search(
-            r"discord(?:app)?.com/api/webhooks/(?P<id>[0-9]{17,20})/(?P<token>[A-Za-z0-9\.\-\_]{60,68})",
+            r"discord(?:app)?.com/api/webhooks/(?P<id>\d{17,20})/(?P<token>[A-z\d.\-_]{60,68})",
```
Comment (Middledot) [inline discord/webhook/sync.py:708]:
```suggestion
            r"discord(?:app)?.com/api/webhooks/(?P<id>\d{17,20})/(?P<token>[\w\.\-_]{60,68})",
```
These could be shortened

File: discord/webhook/async_.py [hunk 2]
```diff
@@ -1136,7 +1136,7 @@ def from_url(
             A partial webhook is just a webhook object with an ID and a token.
         """
         m = re.search(
-            r"discord(?:app)?.com/api/webhooks/(?P<id>[0-9]{17,20})/(?P<token>[A-Za-z0-9\.\-\_]{60,68})",
+            r"discord(?:app)?.com/api/webhooks/(?P<id>\d{17,20})/(?P<token>[A-z\d.\-_]{60,68})",
```
Comment (Middledot) [inline discord/webhook/async_.py:1139]:
```suggestion
            r"discord(?:app)?.com/api/webhooks/(?P<id>\d{17,20})/(?P<token>[\w\.\-_]{60,68})",
```

File: discord/ext/commands/converter.py [hunk 3]
```diff
@@ -137,7 +137,7 @@ async def convert(self, ctx: Context, argument: str) -> T_co:
         raise NotImplementedError("Derived classes need to implement this.")
 
 
-_ID_REGEX = re.compile(r"([0-9]{15,20})$")
+_ID_REGEX = re.compile(r"(\d{15,20})$")
```
Comment (Middledot) [inline discord/ext/commands/converter.py:140]:
```suggestion
_ID_REGEX = re.compile(r"([0-9]{15,20})$")
```

File: discord/ext/commands/converter.py [hunk 4]
```diff
@@ -160,7 +160,7 @@ class ObjectConverter(IDConverter[discord.Object]):
     """
 
     async def convert(self, ctx: Context, argument: str) -> discord.Object:
-        match = self._get_id_match(argument) or re.match(r"<(?:@(?:!|&)?|#)([0-9]{15,20})>$", argument)
+        match = self._get_id_match(argument) or re.match(r"<(?:@[!&]?|#)(\d{15,20})>$", argument)
```
Comment (Middledot) [inline discord/ext/commands/converter.py:163]:
```suggestion
        match = self._get_id_match(argument) or re.match(r"<(?:@[!&]?|#)([0-9]{15,20})>$", argument)
```

File: discord/ext/commands/converter.py [hunk 5]
```diff
@@ -225,7 +225,7 @@ async def query_member_by_id(self, bot, guild, user_id):
 
     async def convert(self, ctx: Context, argument: str) -> discord.Member:
         bot = ctx.bot
-        match = self._get_id_match(argument) or re.match(r"<@!?([0-9]{15,20})>$", argument)
+        match = self._get_id_match(argument) or re.match(r"<@!?(\d{15,20})>$", argument)
```
Comment (Middledot) [inline discord/ext/commands/converter.py:228]:
```suggestion
        match = self._get_id_match(argument) or re.match(r"<@!?([0-9]{15,20})>$", argument)
```

File: discord/ext/commands/converter.py [hunk 6]
```diff
@@ -280,7 +280,7 @@ class UserConverter(IDConverter[discord.User]):
     """
 
     async def convert(self, ctx: Context, argument: str) -> discord.User:
-        match = self._get_id_match(argument) or re.match(r"<@!?([0-9]{15,20})>$", argument)
+        match = self._get_id_match(argument) or re.match(r"<@!?(\d{15,20})>$", argument)
```
Comment (Middledot) [inline discord/ext/commands/converter.py:283]:
```suggestion
        match = self._get_id_match(argument) or re.match(r"<@!?([0-9]{15,20})>$", argument)
```

File: discord/ext/commands/converter.py [hunk 7]
```diff
@@ -336,11 +336,11 @@ class PartialMessageConverter(Converter[discord.PartialMessage]):
 
     @staticmethod
     def _get_id_matches(ctx, argument):
-        id_regex = re.compile(r"(?:(?P<channel_id>[0-9]{15,20})-)?(?P<message_id>[0-9]{15,20})$")
+        id_regex = re.compile(r"(?:(?P<channel_id>\d{15,20})-)?(?P<message_id>\d{15,20})$")
```
Comment (Middledot) [inline discord/ext/commands/converter.py:339]:
```suggestion
        id_regex = re.compile(r"(?:(?P<channel_id>[0-9]{15,20})-)?(?P<message_id>[0-9]{15,20})$")
```

File: discord/ext/commands/converter.py [hunk 8]
```diff
@@ -336,11 +336,11 @@ class PartialMessageConverter(Converter[discord.PartialMessage]):
...
         link_regex = re.compile(
             r"https?://(?:(ptb|canary|www)\.)?discord(?:app)?\.com/channels/"
-            r"(?P<guild_id>[0-9]{15,20}|@me)"
-            r"/(?P<channel_id>[0-9]{15,20})/(?P<message_id>[0-9]{15,20})/?$"
+            r"(?P<guild_id>\d{15,20}|@me)"
+            r"/(?P<channel_id>\d{15,20})/(?P<message_id>\d{15,20})/?$"
```
Comment (Middledot) [inline discord/ext/commands/converter.py:343]:
```suggestion
            r"(?P<guild_id>[0-9]{15,20}|@me)"
            r"/(?P<channel_id>[0-9]{15,20})/(?P<message_id>[0-9]{15,20})/?$"
```

File: discord/ext/commands/converter.py [hunk 9]
```diff
@@ -433,7 +433,7 @@ async def convert(self, ctx: Context, argument: str) -> discord.abc.GuildChannel
     def _resolve_channel(ctx: Context, argument: str, attribute: str, type: Type[CT]) -> CT:
         bot = ctx.bot
 
-        match = IDConverter._get_id_match(argument) or re.match(r"<#([0-9]{15,20})>$", argument)
+        match = IDConverter._get_id_match(argument) or re.match(r"<#(\d{15,20})>$", argument)
```
Comment (Middledot) [inline discord/ext/commands/converter.py:436]:
```suggestion
        match = IDConverter._get_id_match(argument) or re.match(r"<#([0-9]{15,20})>$", argument)
```

File: discord/ext/commands/converter.py [hunk 10]
```diff
@@ -464,7 +464,7 @@ def check(c):
     def _resolve_thread(ctx: Context, argument: str, attribute: str, type: Type[TT]) -> TT:
         bot = ctx.bot
 
-        match = IDConverter._get_id_match(argument) or re.match(r"<#([0-9]{15,20})>$", argument)
+        match = IDConverter._get_id_match(argument) or re.match(r"<#(\d{15,20})>$", argument)
```
Comment (Middledot) [inline discord/ext/commands/converter.py:467]:
```suggestion
        match = IDConverter._get_id_match(argument) or re.match(r"<#([0-9]{15,20})>$", argument)
```

File: discord/ext/commands/converter.py [hunk 11]
```diff
@@ -607,7 +607,7 @@ class ColourConverter(Converter[discord.Colour]):
         Added support for ``rgb`` function and 3-digit hex shortcuts
     """
 
-    RGB_REGEX = re.compile(r"rgb\s*\((?P<r>[0-9]{1,3}%?)\s*,\s*(?P<g>[0-9]{1,3}%?)\s*,\s*(?P<b>[0-9]{1,3}%?)\s*\)")
+    RGB_REGEX = re.compile(r"rgb\s*\((?P<r>\d{1,3}%?)\s*,\s*(?P<g>\d{1,3}%?)\s*,\s*(?P<b>\d{1,3}%?)\s*\)")
```
Comment (Middledot) [inline discord/ext/commands/converter.py:610]:
```suggestion
    RGB_REGEX = re.compile(r"rgb\s*\((?P<r>[0-9]{1,3}%?)\s*,\s*(?P<g>[0-9]{1,3}%?)\s*,\s*(?P<b>[0-9]{1,3}%?)\s*\)")
```

File: discord/ext/commands/converter.py [hunk 12]
```diff
@@ -688,7 +688,7 @@ async def convert(self, ctx: Context, argument: str) -> discord.Role:
         if not guild:
             raise NoPrivateMessage()
 
-        match = self._get_id_match(argument) or re.match(r"<@&([0-9]{15,20})>$", argument)
+        match = self._get_id_match(argument) or re.match(r"<@&(\d{15,20})>$", argument)
```
Comment (Middledot) [inline discord/ext/commands/converter.py:691]:
```suggestion
        match = self._get_id_match(argument) or re.match(r"<@&([0-9]{15,20})>$", argument)
```

File: discord/ext/commands/converter.py [hunk 13]
```diff
@@ -767,7 +767,7 @@ class EmojiConverter(IDConverter[discord.Emoji]):
     """
 
     async def convert(self, ctx: Context, argument: str) -> discord.Emoji:
-        match = self._get_id_match(argument) or re.match(r"<a?:[a-zA-Z0-9\_]{1,32}:([0-9]{15,20})>$", argument)
+        match = self._get_id_match(argument) or re.match(r"<a?:\w{1,32}:(\d{15,20})>$", argument)
```
Comment (Middledot) [inline discord/ext/commands/converter.py:770]:
```suggestion
        match = self._get_id_match(argument) or re.match(r"<a?:\w{1,32}:([0-9]{15,20})>$", argument)
```

File: discord/ext/commands/converter.py [hunk 14]
```diff
@@ -801,7 +801,7 @@ class PartialEmojiConverter(Converter[discord.PartialEmoji]):
     """
 
     async def convert(self, ctx: Context, argument: str) -> discord.PartialEmoji:
-        match = re.match(r"<(a?):([a-zA-Z0-9\_]{1,32}):([0-9]{15,20})>$", argument)
+        match = re.match(r"<(a?):(\w{1,32}):(\d{15,20})>$", argument)
```
Comment (Middledot) [inline discord/ext/commands/converter.py:804]:
```suggestion
        match = re.match(r"<(a?):(\w{1,32}):([0-9]{15,20})>$", argument)
```

File: discord/ext/commands/converter.py [hunk 15]
```diff
@@ -936,7 +936,7 @@ def repl(match: re.Match) -> str:
             transformed = transforms[type](id)
             return transformed
 
-        result = re.sub(r"<(@[!&]?|#)([0-9]{15,20})>", repl, argument)
+        result = re.sub(r"<(@[!&]?|#)(\d{15,20})>", repl, argument)
```
Comment (Middledot) [inline discord/ext/commands/converter.py:939]:
```suggestion
        result = re.sub(r"<(@[!&]?|#)([0-9]{15,20})>", repl, argument)
```

File: discord/partial_emoji.py [hunk 16]
```diff
@@ -93,7 +93,7 @@ class PartialEmoji(_EmojiTag, AssetMixin):
 
     __slots__ = ("animated", "name", "id", "_state")
 
-    _CUSTOM_EMOJI_RE = re.compile(r"<?(?P<animated>a)?:?(?P<name>[A-Za-z0-9\_]+):(?P<id>[0-9]{13,20})>?")
+    _CUSTOM_EMOJI_RE = re.compile(r"<?(?P<animated>a)?:?(?P<name>\w+):(?P<id>\d{13,20})>?")
```
Comment (Middledot) [inline discord/partial_emoji.py:96]:
```suggestion
    _CUSTOM_EMOJI_RE = re.compile(r"<?(?P<animated>a)?:?(?P<name>\w+):(?P<id>[0-9]{13,20})>?")
```

File: discord/utils.py [hunk 17]
```diff
@@ -862,7 +862,7 @@ def escape_mentions(text: str) -> str:
     :class:`str`
         The text with the mentions removed.
     """
-    return re.sub(r"@(everyone|here|[!&]?[0-9]{17,20})", "@\u200b\\1", text)
+    return re.sub(r"@(everyone|here|[!&]?\d{17,20})", "@\u200b\\1", text)
```
Comment (Middledot) [inline discord/utils.py:865]:
```suggestion
    return re.sub(r"@(everyone|here|[!&]?[0-9]{17,20})", "@\u200b\\1", text)
```

**Review by Middledot — COMMENTED** [review]
`\d` matches `0-9` as well as unicode digits like `٠١٢٣٤٥٦٧٨٩`. It would be better to keep it stricter.
Otherwise looks good

**Review by Lulalaby — APPROVED** [review]

**Review by Middledot — APPROVED** [review]

**Pull request merged**